Tests/FluentAssertions.Specs/FindAssembly.cs

```diff
@@ -1,11 +1,46 @@
-﻿using System.Reflection;
+﻿using System;
+using System.Reflection;
 
 namespace FluentAssertions.Specs;
 
 public static class FindAssembly
 {
-    public static Assembly Containing<T>()
+    public static Assembly Containing<T>() => typeof(T).Assembly;
+
+    public static Assembly Stub(string publicKey = "") => new AssemblyStub(publicKey);
+
+    private sealed class AssemblyStub : Assembly
     {
-        return typeof(T).Assembly;
+        private readonly AssemblyName assemblyName = new();
+
+        public override string FullName => nameof(AssemblyStub);
+
+        public AssemblyStub(string publicKey)
+        {
+            assemblyName.SetPublicKey(FromHexString(publicKey));
+        }
+
+        public override AssemblyName GetName() => assemblyName;
+
+#if NET6_0_OR_GREATER
+        private static byte[] FromHexString(string chars)
+            => string.IsNullOrEmpty(chars)
+            ? Array.Empty<byte>()
+            : Convert.FromHexString(chars);
+#else
+    private static byte[] FromHexString(string chars)
+    {
+        var bytes = new byte[chars.Length / 2];
+
+        for (int i = 0, j = 0; j < bytes.Length; j++, i += 2)
+        {
+            bytes[j] = (byte)((Hex(chars[i]) << 4) | Hex(chars[i + 1]));
+
+            static byte Hex(char c) => (byte)(c > '9' ? (c & 0x5F) - 55 : c - '0');
+        }
+
+        return bytes;
+    }
```

Comment:
🔧 I found this implementation a bit hard to read, so I asked ChatGPT to provide a suggestion:

```
        if (hexString.Length % 2 != 0)
        {
            throw new ArgumentException("Input string length must be even.");
        }

        byte[] byteArray = new byte[hexString.Length / 2];

        for (int i = 0; i < hexString.Length; i += 2)
        {
            string hexByte = hexString.Substring(i, 2);
            byte value = Convert.ToByte(hexByte, 16);
            byteArray[i / 2] = value;
        }

        return byteArray;
```

Reply:
I suggested the bit hacking implementation.
I perhaps sacrificed to much readability in my quest to avoid allocations 😅